tests/drivers/can/api/src/classic.c

```diff
@@ -615,6 +663,17 @@ ZTEST(can_classic, test_add_filter)
 	can_remove_rx_filter(can_dev, filter_id);
 }
 
+/**
+ * @brief Test adding filter without callback.
+ */
+ZTEST(can_classic, test_add_filter_without_callback)
```

Comment:
There is a set of negative tests which check that can_add_rx_filter() returns -EINVAL when filter is incorrect:
https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/drivers/can/api/src/classic.c#L637
Here, I check that error code is returned when user tries to set filter without callback function.